src/vsr/replica.zig

```diff
@@ -475,6 +481,17 @@ pub fn ReplicaType(
                 }
                 assert(self.op_head_certain());
 
+                // Solo replicas must increment their view after recovery.
+                // Otherwise, two different versions of an op could exist within a single view
+                // (the former version truncated as a torn write).
+                self.log_view += 1;
+                self.view += 1;
+                self.view_headers.array.len = 0;
+                self.view_headers.array.appendAssumeCapacity(
+                    self.journal.header_with_op(self.op).?.*,
+                );
+                self.view_durable_update();
```

Comment:
https://github.com/tigerbeetledb/tigerbeetle/pull/561/commits/b3457197dc2ec00fa3d39146180a269476208e81 for more detail.